File: services/gateway/app/api/evidence.py
from fastapi import APIRouter


router = APIRouter()


_EVIDENCE = {
    "inc-001": {
        "timeline": [
            {"time": "14:00", "event": "Alert received", "source": "AlertManager"},
            {"time": "14:01", "event": "Investigation started", "source": "Investigation Agent"},
            {"time": "14:03", "event": "Evidence collected", "source": "Kubernetes MCP"},
        ],
        "items": [
            {"type": "metric", "name": "container_memory_usage", "confidence": 0.92},
            {"type": "event", "name": "OOMKilled", "confidence": 0.95},
        ],
        "rca": {
            "root_cause": "memory pressure caused container OOMKilled",
            "confidence": 0.91,
        },
    }
}


@router.get("/incidents/{incident_id}/evidence")
def get_incident_evidence(incident_id: str):
    return _EVIDENCE.get(
        incident_id,
        {
            "timeline": [],
            "items": [],
            "rca": None,
        },
    )

File: web/lib/api/evidence.ts
export interface EvidenceTimeline {
  time: string;
  event: string;
  source: string;
}

export interface IncidentEvidence {
  timeline: EvidenceTimeline[];
  items: Array<{
    type: string;
    name: string;
    confidence: number;
  }>;
  rca?: {
    root_cause: string;
    confidence: number;
  } | null;
}

const API_BASE = process.env.NEXT_PUBLIC_API_BASE ?? "http://localhost:8000";

export async function getIncidentEvidence(id: string): Promise<IncidentEvidence> {
  const response = await fetch(`${API_BASE}/api/incidents/${id}/evidence`, {
    cache: "no-store",
  });

  if (!response.ok) {
    throw new Error("failed to fetch evidence");
  }

  return response.json();
}
